crates/circuit/src/circuit_data.rs

```diff
@@ -559,22 +560,34 @@ impl CircuitData {
     /// Get a (cached) sorted list of the Python-space `Parameter` instances tracked by this circuit
     /// data's parameter table.
     #[getter]
-    pub fn get_parameters<'py>(&self, py: Python<'py>) -> Bound<'py, PyList> {
-        self.param_table.py_parameters(py)
+    pub fn get_parameters<'py>(&self, py: Python<'py>) -> PyResult<Bound<'py, PyList>> {
+        PyList::new(
+            py,
+            self.param_table
+                .parameters()
+                .iter()
+                .map(|symbol| symbol.coerce_into_py(py).unwrap()),
+        )
     }
 
     pub fn unsorted_parameters<'py>(&self, py: Python<'py>) -> PyResult<Bound<'py, PySet>> {
-        self.param_table.py_parameters_unsorted(py)
+        PySet::new(
+            py,
+            self.param_table
+                .parameters_unsorted()
+                .iter()
+                .map(|symbol| symbol.coerce_into_py(py).unwrap()),
+        )
```

Comment:
This now involves two allocations one for the `HashSet` inside `parameters_unsorted()` and the pyset here. Couldn't you just do:

```suggestion
        PySet::new(
            py,
            self.param_table
                .parameters_by_uuid()
                .iter()
                .map(|symbol| symbol.coerce_into_py(py).unwrap()),
        )
```
So we just do the set collection once, just in python. I don't see a method to just get `ParameterTable::by_uuid.values()` directly so we need to add something that returns just a reference or an iterator that doesn't collect.

Reply:
Oh good point, I'll add a `iter_parameters` to iterate (in unsorted order)